### PR TITLE
Added c,python and php implementation of 0-1 knapsack problem

### PR DESCRIPTION
**Pull request by Jaseemck**
Added solution 0-1 knapsack problem in c , python and php language. It is based on Dynamic Programming. Detailed explanation of what the problem is and what dynamic programming is, is commented the mentioned program files. 
A test case and corresponding output is given in the program for testing.
Alter the inputs and check the outputs in all cases.
Happy Coding!

### CODE REVIEW

**Comment by Jaseemck** [comment]
Please merge the pull request.

**Comment by jainaman224** [comment]
Please fix file names of python and php code.
ex:- `Knapsack.py`.

**Comment by Jaseemck** [comment]
Done....

**Comment by jainaman224** [comment]
@Jaseemck Please use better variable names.

File: Knapsack/Knapsack.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+// Knapsack function based on Dynamic Programming 
+/*
+Variables:
+mw-maximum weight
+wt[]-Array of weights
+val[]-Array of values
+n-number of values
+*/
+int knapSack(int mw, int wt[], int val[], int n)
```
Comment (jainaman224) [inline Knapsack/Knapsack.c]: Instead of mw use Max weight.

File: Knapsack/Knapsack.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Variables:
+mw-maximum weight
+wt[]-Array of weights
+val[]-Array of values
+n-number of values
+*/
+int knapSack(int mw, int wt[], int val[], int n) 
+{ 
+    int i, j; 
+    int K[n+1][mw+1];
```
Comment (jainaman224) [inline Knapsack/Knapsack.c]: Use something better than K. Which is easy to understand.

File: Knapsack/Knapsack.c [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    int K[n+1][mw+1]; 
+
+
+    for (i = 0; i <= n; i++) 
+    { 
+	for (j = 0; j <= mw; j++) 
+	{ 
+		if (i==0 || j==0) 
+			K[i][j] = 0; 
+		else if (wt[i-1] <= j)
```
Comment (jainaman224) [inline Knapsack/Knapsack.c]: Instead of wt use weights.

**Comment by jainaman224** [comment]
@Jaseemck Please remove a.out file as well.
Then I will merge.

**Pull request merged**